sprocs/users_randomly_generate.sql

```diff
@@ -0,0 +1,60 @@
+
+    first_names text[];
+
+    status int;
+    name text;
+    email text;
+    uin text;
+
+    _user_id bigint;
+
+BEGIN
+    FOR i IN 1..count LOOP
+
+        name := '';
+        status := FLOOR(RANDOM() * 10);
+        --RAISE NOTICE 'Status is %', status;
+
+        IF MOD(status, 2) = 1 THEN
+            first_names = boy_names;
+        ELSE
+            first_names = girl_names;
+        END IF;
+
+        -- 20% chance they have multiple first names
+        IF status >= 8 THEN
+            name = name || first_names[FLOOR((RANDOM()*ARRAY_LENGTH(first_names, 1)))::int] || ' ';
+        END IF;
+
+        name = name || first_names[FLOOR((RANDOM()*ARRAY_LENGTH(first_names, 1)))::int] || ' ';
+        name = name || surnames[FLOOR((RANDOM()*ARRAY_LENGTH(surnames, 1)))::int];
+
+        email = LOWER(name);
+        email = REPLACE(email, ' ', '.') || '@example.com';
+        --RAISE NOTICE 'Name is %', name;
+        --RAISE NOTICE 'Email is %', email;
+
+        INSERT INTO users (name, uid) VALUES (name, email)
+        RETURNING users.user_id INTO _user_id;
+
+        course_instance_id = NULLIF(course_instance_id, '');
```

Comment:
```suggestion
```

I think this is unnecessary if we make the argument a `bigint`?

Reply:
`null` isn't a valid int, so you can't pass it in. Passing in a string and converting it as needed seems to be the way to work around that? Maybe you have a different pattern?

Reply:
Most of that complexity is around making the SQL 1) generate random users and 2) optionally enroll them in a specified course instance. If we made the enrollment not optional it might simplify things?

Reply:
In SQL I believe `NULL` is a valid instance of any base type. The problem is that we get an empty string (not null) when the `course_instance_id` input field is blank. I changed the code so that the underlying SQL function takes the correct types (int/bigint) and put the `nullif()` in the `admin_query` SQL query.